include/zephyr/net/conn_mgr.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_CONN_MGR_H_
+#define ZEPHYR_INCLUDE_CONN_MGR_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(CONFIG_NET_CONNECTION_MANAGER)
+
+struct net_if;
+struct net_l2;
+
+/**
+ * @brief Resend either NET_L4_CONNECTED or NET_L4_DISCONNECTED depending on whether connectivity
+ * is currently available.
+ */
+void conn_mgr_resend_status(void);
+
+/**
+ * @brief Mark an iface to be ignored by conn_mgr.
+ *
+ * Ignoring an iface forces conn_mgr to consider it unready/disconnected.
+ *
+ * This means that events related to the iface connecting/disconnecting will not be fired,
+ * and if the iface was connected before being ignored, events will be fired as though it
+ * disconnected at that moment.
+ *
+ * @param iface - iface to be ignored.
+ */
+void conn_mgr_ignore_iface(struct net_if *iface);
+
+/**
+ * @brief Stop ignoring an iface.
+ *
+ * conn_mgr will no longer be forced to consider the iface unreadly/disconnected.
+ *
+ * Events related to the iface connecting/disconnecting will no longer be blocked,
+ * and if the iface was connected before being unignored, events will be fired as though
+ * it connected in that moment.
+ *
+ * @param iface - iface to no longer ignore.
+ */
+void conn_mgr_unignore_iface(struct net_if *iface);
```

Comment:
This unignore word looks weird. I was not able to find it in dictionaries so it looks like it is not proper English. Dunno what we should have here, Perhaps `conn_mgr_start_ignore_iface()` and `conn_mgr_stop_ignore_iface()` although they are a bit longish.

Reply:
What about `conn_mgr_ignore_iface` and `conn_mgr_watch_iface`? This would align better with the documentation I wrote anyways. @rlubos @wbober You also brought up the naming of these two functions, would this naming be satisfying to you as well?

Reply:
I've gone ahead and force-pushed `ignore/unignore->ignore/watch` on the assumption/suspicion that everyone involved will either not care or prefer it. Let me know if this is satisfactory

Reply:
Ignore/watch naming LGTM.

Reply:
Sounds fine to me